Chess/desk.cpp

```diff
@@ -42,8 +45,123 @@ Desk::Desk(QObject *parent) :
     for (int i=0; i<8; i++) {
         buffer_[6][i] = new Pawn(this, fcolor);
     }
+}
+
+
+bool Desk::grabFigure(const Cell &cell)
+{
+    Figure *curr_fig = getFigure(cell);
+
+    if (curr_fig == NULL)
+        return false;

```

Comment:
You can either use return codes or exceptions, using bouth in the same component/project/sub-project is considered to be a bad stile